runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -599,6 +602,36 @@ impl<Consensus: modules::consensus::API> Module<Consensus> {
         state::get_undelegations(args.to)
     }
 
+    #[handler(call = "consensus.SharesToTokens", internal)]
+    fn internal_shares_to_tokens<C: Context>(
+        ctx: &C,
+        args: types::SharesToTokens,
+    ) -> Result<u128, Error> {
+        let params = Self::params();
+        <C::Runtime as Runtime>::Core::use_tx_gas(params.gas_costs.shares_to_tokens)?;
+
+        let account = Consensus::account(ctx, args.address)?;
+        let pool = match args.pool {
+            types::SharePool::Active => &account.escrow.active,
+            types::SharePool::Debonding => &account.escrow.debonding,
+            _ => return Err(Error::InvalidArgument),
+        };
+
+        let tokens = args
+            .shares
+            .checked_mul(
+                u128::try_from(pool.balance.clone()).expect("pool balance should fit into u128"),
```

Comment:
I would avoid crashing the runtime in these cases and would instead just return an invalid argument error.